packages/core/src/plugin/index.ts

```diff
@@ -30,15 +28,15 @@ export async function resolveFarmPlugins(config: UserConfig) {
 
     if (
       typeof plugin === 'string' ||
-      (isArray(plugin) && typeof plugin[0] === 'string')
+      (Array.isArray(plugin) && typeof plugin[0] === 'string')
     ) {
       rustPlugins.push(
         await rustPluginResolver(plugin as string, config.root ?? process.cwd())
       );
-    } else if (isObject(plugin)) {
+    } else if (typeof plugin === 'object') {
```

Comment:
typeof is not accurate, for example, a null is received

Reply:
@shulandmimi you are correct that `typeof` in js can be misleading, especially when dealing with null values, as `typeof` null returns 'object'.  Can you suggest me what to do when error occur.

Reply:
```suggestion
    } else if (isObject(plugin)) {
```